java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -116,12 +135,66 @@ public boolean isAllowed(@NonNull String httpMethod, @NonNull URL url) {
         return rules.getAllowAllEndpoints() || getEndpoint(httpMethod, url).isPresent();
     }
 
+    @Override
+    public boolean isAllowed(String httpMethod, URL url, @Nullable String contentType,
+            @Nullable String requestBody) {
+        if (rules.getAllowAllEndpoints()) {
+            return true;
+        }
+        Optional<Pair<Pattern, Endpoint>> matchingEndpoint = getEndpoint(httpMethod, url);
+
+        if (matchingEndpoint.isEmpty()) {
+            log.warning(String.format("No endpoint found for %s %s", httpMethod, url));
+            return false;
+        }
+
+        Endpoint endpoint = matchingEndpoint.get().getValue();
+
+        if (endpoint.getRequestBody() == null) {
+            return true;
+        } else {
+            if (endpoint.getRequestBody().getRequired() && StringUtils.isEmpty(requestBody)) {
+                log.warning(String.format("Request body required for %s %s, but none provided",
+                        httpMethod, url));
+                return false;
+            }
+            if (endpoint.getRequestBody().getContent() == null) {
+                // no schemas defined for request body, so we allow it
+                return true;
+            } else if (StringUtils.isEmpty(requestBody)) {
+                // empty request body, and none required, so we allow it
+                return true;
+            } else {
+                // default content-type to application/json (tbh, is this right?? it's just our
+                // historical default)
+                String parsedContentType =
+                        Optional.ofNullable(contentType).orElse("application/json");
```

Comment:
About hardcoding `application/json` and others.. worth to use `ContentType.APPLICATION_JSON.toString()` from `org.apache.http.entity` ?